docs/worker-versioning.md

```diff
@@ -0,0 +1,229 @@
+```
+2. Deploy your worker with new Build ID while keeping the old Build ID still running.
+2. [Use CLI to update Worker Versioning Rules](#update-rules-using-cli) 
+to send new (and possibly existing, if you are deploying a compatible build) executions 
+to the new Build ID. Depending on your deployment strategy you may need to update the 
+rules multiple times.
+3. Use CLI to get reachability status of the old build ID. Once the Build ID is not 
+reachable, decommission the old Build ID.
+
+**Note for Temporal Server Operators:** Worker Versioning needs to be enabled from server
+before being able to run versioned workers or update the rules. To enable Worker Versioning
+set the following dynamic configs to `true`: `frontend.workerVersioningRuleAPIs`, 
+`frontend.workerVersioningWorkflowAPIs`. Both configs are per Task Queue.
+
+
+### Update Rules Using CLI
+You can use `temporal task-queue versioning` commands to update and read the Versioning 
+rules of a given Task Queue. Here are a few examples:
+
+**Add Assignment rule:** send 10% of new executions to Build ID "abc-123".
+```shell
+temporal task-queue versioning insert-assignment-rule --task-queue MY_TQ --build-id abc-123 --percentage 10
+```
+
+**Commit Build ID** to complete the rollout of "abc-123" and cleanup unnecessary rules.
+```shell
+temporal task-queue versioning commit-build-id --task-queue MY_TQ --build-id abc-123
+```
+
+**Add Redirect rule** from "abc-123" to "xyz-789":
+```shell
+temporal task-queue versioning add-redirect-rule --task-queue MY_TQ --source-build-id abc-123 --target-build-id xyz-789
+```
+
+**List Versioning rules**:
+```shell
+temporal task-queue versioning get-rules --task-queue MY_TQ
+```
+
+### Get Reachability Info via CLI
+You can use `temporal task-queue describe` command to get reachability status of a Build ID.
+Example: report reachability of Build ID "abc-123" on Task Queue MY_TQ:
+```shell
+temporal task-queue describe --task-queue MY_TQ --select-build-id abc-123 --report-reachability
+```
+
+The reachability status can be one of the following:
+- **REACHABLE:** Build ID may be used by new workflows or activities (base on versioning 
+rules), or there MAY be open workflows or backlogged activities assigned to it.
+- **CLOSED_WORKFLOWS_ONLY:** Build ID does not have open workflows and is not reachable
+by new workflows, but MAY have closed workflows within the namespace retention period. 
+Not applicable to activity-only task queues.
+- **UNREACHABLE:** Build ID is not used for new executions, nor it has been used by 
+any existing execution within the retention period.
+- **UNSPECIFIED:** Task reachability is not reported
+
+#### Caveats:
+- Task Reachability is eventually consistent; there may be a delay until it converges to the most
+accurate value but it is designed in a way to take the more conservative side until it converges.
+For example REACHABLE is more conservative than CLOSED_WORKFLOWS_ONLY.
```

Comment:
Not sure what this sentence means or how that should affect my usage.

Reply:
When you're testing, you might run a single workflow on a build id, see that it's completed, change the assignment rules, but then reachability for that build id still reports REACHABLE, instead of CLOSED_WORKFLOWS_ONLY as you might expect. This just explains why that is: there's a delay, and it errs on the conservative side. The other way (reporting something less than REACHABLE when there are running workflows, leading an operator to decommission workers), could lead to an outage.